From dda645b7bebd926ea31ebbbdc80902bd2936d67a Mon Sep 17 00:00:00 2001
From: Duncan Macleod <macleoddm@cardiff.ac.uk>
Date: Mon, 5 Oct 2026 12:07:53 +0100
Subject: [PATCH] fix(framecpp-swig/python): use PyLong_FromLong instead of
 PyInt_FromLong

The latter was removed in Python 3, but supported by SWIG until 4.5

Closes https://git.ligo.org/computing/ldastools/LDAS_Tools/-/work_items/416
---
 framecpp-swig/python/source/python/GPSTime.i        | 2 +-
 framecpp-swig/python/source/python/frameCPPMacros.i | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/framecpp-swig/python/source/python/GPSTime.i b/framecpp-swig/python/source/python/GPSTime.i
index d55f29fcd..7501d1366 100644
--- a/framecpp-swig/python/source/python/GPSTime.i
+++ b/framecpp-swig/python/source/python/GPSTime.i
@@ -84,7 +84,7 @@
           ++tuple_index )
     {
       PyTuple_SET_ITEM( tuple, tuple_index,
-                        PyInt_FromLong( *cur ) );
+                        PyLong_FromLong( *cur ) );
     }
   $result = tuple;
  }
diff --git a/framecpp-swig/python/source/python/frameCPPMacros.i b/framecpp-swig/python/source/python/frameCPPMacros.i
index 4d089c6bd..a4c9906bb 100644
--- a/framecpp-swig/python/source/python/frameCPPMacros.i
+++ b/framecpp-swig/python/source/python/frameCPPMacros.i
@@ -121,12 +121,12 @@
 
 %typemap( in ) IFrameFStream::size_type
 {
-    $1 = PyInt_AsLong( $input );
+    $1 = PyLong_AsLong( $input );
 }
 
 %typemap( out ) IFrameFStream::size_type
 {
-    $result = PyInt_FromLong( $1 );
+    $result = PyLong_FromLong( $1 );
 }
 
 //::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
-- 
2.47.3

